src/procfs.cpp

```diff
@@ -137,6 +140,50 @@ uptime procfs::get_uptime() const
     return parsers::parse_uptime_line(line);
 }
 
+sys_stat procfs::get_stat() const
+{
+    static const std::string STAT_FILE("stat");
+    auto path = _root + STAT_FILE;
+
+    std::vector<std::string> lines;
+    parsers::parse_lines(
+        path,
+        std::back_inserter(lines),
+        [&](const std::string &line) { return line; }
+    );
+
+    sys_stat stats;
+
+    for (const std::string& line : lines) {
+        if (line.rfind("cpu", 0) == 0) {
+            stats.cpus.emplace_back(parsers::parse_stat_cpu_line(line));
```

Comment:
Following your suggestion and my comment about the parser, I'm not sure this is a correct assumption. I mean, not sure CPUs are going to be always ordered correctly.
Especially with hot-swapping and such, I wonder how this is going to be handled by the machine...
Maybe we should use the `X` from `cpuX` to add the values to the right index in the vector...

Reply:
I walked over some recent kernel code, and it seems ordered, can you verify me?

https://github.com/torvalds/linux/blob/5bfc75d92efd494db37f5c4c173d3639d4772966/fs/proc/stat.c#L165

```
for_each_online_cpu() ->
#define for_each_online_cpu(cpu)   for_each_cpu((cpu), cpu_online_mask)

```
```
/**
 * for_each_cpu - iterate over every cpu in a mask
 * @cpu: the (optionally unsigned) integer iterator
 * @mask: the cpumask pointer
 *
 * After the loop, cpu is >= nr_cpu_ids.
 */
#define for_each_cpu(cpu, mask)				\
	for ((cpu) = -1;				\
		(cpu) = cpumask_next((cpu), (mask)),	\
		(cpu) < nr_cpu_ids;)


```
```

/**
 * cpumask_next - get the next cpu in a cpumask
 * @n: the cpu prior to the place to search (ie. return will be > @n)
 * @srcp: the cpumask pointer
 *
 * Returns >= nr_cpu_ids if no further cpus set.
 */
unsigned int cpumask_next(int n, const struct cpumask *srcp)
{
	/* -1 is a legal arg here. */
	if (n != -1)
		cpumask_check(n);
	return find_next_bit(cpumask_bits(srcp), nr_cpumask_bits, n + 1);
}
EXPORT_SYMBOL(cpumask_next);
```

Reply:
Looks good. I think we can assume they are ordered till proven otherwise :)